python/BioSimSpace/FreeEnergy/_atm.py

```diff
@@ -0,0 +1,1393 @@
+            if not isinstance(ligand2_rigid_core, list):
+                raise TypeError("ligand2_rigid_core must be a list")
+            elif not len(ligand2_rigid_core) == 3:
+                raise ValueError("ligand2_rigid_core must have length 3")
+
+            # copy the ligands
+            ligand1 = ligand1.copy()
+            move_to_origin(ligand1)
+            ligand2 = ligand2.copy()
+            move_to_origin(ligand2)
+
+        pre_translation_lig1_core_coords = []
+
+        for i in ligand1_rigid_core:
+            x = ligand1.getAtoms()[i].coordinates().x().value()
+            y = ligand1.getAtoms()[i].coordinates().y().value()
+            z = ligand1.getAtoms()[i].coordinates().z().value()
+            pre_translation_lig1_core_coords.append((x, y, z))
+
+        point1, point2, distance = furthest_points(pre_translation_lig1_core_coords)
+        vector = vector_from_points(point1, point2)
+
+        # need to know the size of ligand1
+        lig1_coords = []
+        for i in ligand1.getAtoms():
+            x = i.coordinates().x().value()
+            y = i.coordinates().y().value()
+            z = i.coordinates().z().value()
+            lig1_coords.append((x, y, z))
+
+        lig1_point1, lig1_point2, lig1_distance = furthest_points(lig1_coords)
+
+        # Translate ligand2 so they don't overlap
+        ligand2.translate(
+            [
+                lig1_distance * 2 * vector[0],
+                lig1_distance * 2 * vector[1],
+                lig1_distance * 2 * vector[2],
+            ]
+        )
+        # Get coords of rigid core atoms
+        ligand1_core_coords = []
+        ligand2_core_coords = []
+        for i in ligand1_rigid_core:
+            ligand1_core_coords.append(ligand1.getAtoms()[i].coordinates())
+        for i in ligand2_rigid_core:
+            ligand2_core_coords.append(ligand2.getAtoms()[i].coordinates())
+
+        # Create molecule containing both ligands
+        mol = ligand1 + ligand2
+
+        # Create view
+        view = _View(mol)
+
+        # Create nglview object
+        ngl = view.system(mol)
+        colours = [[1, 1, 0], [1, 0, 1], [0, 1, 1]]
+        # Add spheres to rigid core locations
+        for coord1, coord2, colour in zip(
+            ligand1_core_coords, ligand2_core_coords, colours
+        ):
+            ngl.shape.add_sphere(
+                [coord1.x().value(), coord1.y().value(), coord1.z().value()],
+                colour,
+                0.7,
+            )
+            ngl.shape.add_sphere(
+                [coord2.x().value(), coord2.y().value(), coord2.z().value()],
+                colour,
+                0.7,
+            )
+        if system is not None:
+            del local_s
+        return ngl
+
+    @staticmethod
+    def run(
+        system,
+        protocol,
+        platform="CPU",
+        work_dir=None,
+        setup_only=False,
+        property_map={},
+    ):
```

Comment:
This is behaving like the constructor of `BioSImSpace.FreeEnergy.Relative`. (It basically constructs a similar object, then calls `.run` on it.) I'll have a think about how best to do this. Here `BioSimSpace.FreeEnregy.AToM` basically does all the setup, whereas the additional `BioSImSpace.FreeEnergy._relativeATM` (which is created by this `run` method) runs the actual simulation.

If possible, it might be better to have `BioSimSpace.AToM.prepare` and `BioSImSpace.AToM.viewRigidCores` as standalone functions, then `BoSimSpace.FreeEnergy.AToM` as the main class that can setup, run, and analyse the simulation, as is done for `BioSimSpace.FreeEnergy.Relative. I understand that `BioSimSpace.FreeEnergy.AToM` stores a lot of state, which simplifies the call to `.prepare`, etc. It might be possible to just change the class to `AToMSetup`, or just `Setup`, to make clear what it's being used for. Will have a think.